### PR TITLE
Update google-api-core

### PR DESCRIPTION
**Pull request by wesleybowman**
###### Motivation for this change

Update the python library `google-api-core` to latest version

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -5308,7 +5308,7 @@ in {
   protobuf = callPackage ../development/python-modules/protobuf {
     disabled = isPyPy;
     doCheck = !isPy3k;
-    protobuf = pkgs.protobuf3_8;
+    protobuf = pkgs.protobuf3_12;
```
Comment (prusnak) [inline pkgs/top-level/python-packages.nix:5311]: See https://github.com/NixOS/nixpkgs/pull/91446 for the protobuf version discussion.

**Comment by jonringer** [comment]
please address conflicts

**Comment by wesleybowman** [comment]
@jonringer Done!

**Comment by jonringer** [comment]
unfortunately, we already have more up-to-date versions in staging:
```
/nix/store/43i4y6jnqg2rkilic1wmbc9cm6sn95jy-python3.8-google-api-core-1.22.1
[09:40:41] jon@jon-desktop /home/jon/projects/nixpkgs (python-unstable)
$ nix-build -A python3Packages.google_auth
/nix/store/q2s0kw1h5d45n1myyfwnvkr3h5i6m2vs-python3.8-google-auth-1.20.1
[09:40:51] jon@jon-desktop /home/jon/projects/nixpkgs (python-unstable)
$
```
this will make their way to master in ~2 weeks

**Comment by wesleybowman** [comment]
@jonringer Ah, nice! I think we can close this PR then.

**Pull request closed without merging**